assignments/week_1/exercise_1_2/exercise_1_2.cpp

```diff
@@ -0,0 +1,34 @@
+// Iterative C++ program to reverse an array and check if string is palindrome
+#include <stdio.h>
+#include <string.h>
+#include <iostream>
+
+int main()
+{
+    // Part 1: Array Reversal
+    int arr[] = {1, 2, 3, 4, 5, 6};
+    int n = sizeof(arr) / sizeof(arr[0]);
+
+    // Print original array
+    std::cout << "Original Array: ";
+    for (int i = 0; i < n; i++)
+        std::cout << arr[i] << " ";
```

Comment:
I think these should be range-based for loops, since we're iterating over the entire array. This would also eliminate the 
 one-letter variables.

```suggestion
    for (int elem : arr)
        std::cout << elem << ' ';
```